README.md

```diff
@@ -63,6 +63,23 @@
   ./spin.py destroy [dev|stage]
 ~~~
 
-#### Notes
+
 
 -   Do not use vagrant directly to spin and environment. It might work, but most likely it won't
+==============================================================================================
+Steps for Docker:
+Setting the App 
+ 1. install docker or docker for windows 
+ 2. Run "docker build -t spinner ." in the directory where docker file is located
+ 3. Run "docker run -t -d --name spincontainer -v /var/run/docker.sock:/var/run/docker.sock -v  {devops_path}:/resources spinner"
```

Comment:
What is {devops_path}? 
```➜ devopsloft git:(leonicus-dockerdev) docker run -t -d --name spincontainer -v /var/run/docker.sock:/var/run/docker.sock -v  {devops_path}:/resources spinner
docker: Error response from daemon: create {devops_path}: "{devops_path}" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.```

Reply:
@lioramilbaum  the meaning is the path to the main folder. Like C:\devopsloft. We can change it to a more descriptive name

Reply:
Can't we omit the '{devops_path:/' and use a relative path? Just 'resources'.

Reply:
> Can't we omit the '{devops_path:/' and use a relative path? Just 'resources'.

No, /resources is the path which will be created in the container. The {devopspath} has to be the full path to the resource on your pc. Relative paths will not work here, since we are using an intermediate container to raise the app container. So it has no way to get a relative path, it has to be told exactly to which folder to mount

Reply:
what is the purpose of mounting the volume?  Can you use a COPY instruction instead of a mount?

Reply:
Additional options - https://github.com/moby/moby/issues/4830

Reply:
I was trying to keep the implementation as was previously, according to the docker-compose.yml, in which these are mounted volumes. If we do not need to persist anything there, COPY is definitely more economical. Can we change it?

Reply:
For now, do whatever is needed to complete the target task. Which is, the ability to spin a DEV environment in all common platforms: win, os-x, linux.
Technical dept is more than welcomed.

Reply:
@leonicus I would aim to make docker work simply with the following:
`docker run --rm -v /var/run/docker.sock:/var/run/docker.sock spinner`
much simpler, then you also don't need the exec into the container and run anything. better practice, less error prone
it *does* means that there is a need for separate instructions to whomever is *developing* the spin-up.
Plus, the relative path is not possible not due to being intermediate container, rather a limitation of docker itself. docker-compose however, allows you to use relative paths

Reply:
@lioramilbaum Well, that part s completed hopefully, though I can't test it on all platforms :) (assuming you replace {devopspath} by C;\devopsloft or whichever path it is on you computer). But I agree that if mounts are not needed, we should change it, I just don;t feel comfortable enough with the architecture to decide that by myself...

Reply:
Then, feel free to consult over the issue/PR/Slack. There are more the enough people who want to assist/guide and make this challenge resolved

Reply:
@ShimiTaNaka the mounting of /var/sock is already there. But that doesn't solve the volumes issue, I checked it. There is a need for an intermediate code that resolves a path on the intermediate container to a path on the host OS...at least that's the solution that I found, and that's the purpose of in_docker.py file

Reply:
@ShimiTaNaka I agree, that's a good aim :) but as long as we want to mount the volumes, it's not possible ( the mounting of /var/sock/docker.sock is already there in my solution, but it requires more code to translate to files in the host OS, docker can't mount from container to container, I checked). So, the remaining questions is, can we COPY instead of mounting, as Liora suggested? Do we need something to persist in those volumes?

Reply:
@ShimiTaNaka, @lioramilbaum, So I'll go ahead and implement as copy, so we can move forward?

Reply:
I am not deep into the design as @ShimiTaNaka, but, I think the end result is currently more important than the technical dept created here.  Go ahead.

Reply:
OK, so it's implemented as copy now. Working on my PC

Reply:
BTW, we should take into account that since we are not mounting volumes, changes on the host OS file will have to be copied manually to container. Meaning, if I'm working on a feature, I'll have to run docker cp to the files I've changed in order to see the changes in the dev environment.